src/core/scalar/vector.rs

```diff
@@ -232,6 +232,16 @@ impl<T: NumEx> Vector<T> for XY<T> {
             y: self.y.max(other.y),
         }
     }
+
+    #[inline]
+    fn clamp(self, min: Self, max: Self) -> Self {
+        glam_assert!(min.x <= max.x);
+        glam_assert!(min.y <= max.y);
+        Self {
+            x: self.x.max(min.x).min(max.x), // TODO: use std::f32::clamp when we update the minimal required rustc version to >= 1.50
```

Comment:
Possibly we don't want to use Rust clamp long term as it will definitely assert/panic. If not following the convention of Rust's clamp it would be worth calling out in the method docs.